pkgs/applications/misc/waybar/default.nix

```diff
@@ -0,0 +1,54 @@
+{ 
+stdenv, fetchFromGitHub, makeWrapper, meson, pkgconfig, ninja, 
+gtkmm3, jsoncpp, libinput, libsigcxx, wayland, wlroots, fmt
+, traySupport  ? true,  libdbusmenu-gtk3 ? null
+, pulseSupport ? false, libpulseaudio    ? null
+, nlSupport    ? true,  libnl            ? null
+, swaySupport  ? false, sway             ? null
+}:
+
+assert traySupport -> libdbusmenu-gtk3 != null;
+assert pulseSupport  -> libpulseaudio != null;
+assert nlSupport -> libnl != null;
+assert swaySupport -> sway != null;
+
+stdenv.mkDerivation rec {
+    name = "waybar-${version}";
+    version = "0.2.3";
+    src = fetchFromGitHub {
+      owner = "Alexays";
+      repo = "Waybar";
+      rev = version;
+      sha256 = "1b0lx4klmwsipni7h566rb89rpqn4lhbysvsyai3gjk6g4fb1k30";
+    };
+
+    patches = [ ./sysconfdir.patch ];
+
+    meta = with stdenv.lib; {
+      description = "Highly customizable Wayland bar for Sway and Wlroots based compositors.";
+      license = licenses.mit;
+      maintainers = [ maintainers.florianfranzen ];
+      platforms = platforms.unix;
+    };
+
+    buildInputs = [ gtkmm3 jsoncpp libinput libsigcxx wayland wlroots fmt ]
+      ++ stdenv.lib.optional  traySupport  libdbusmenu-gtk3
+      ++ stdenv.lib.optional  pulseSupport libpulseaudio
+      ++ stdenv.lib.optional  nlSupport    libnl
+      ++ stdenv.lib.optionals swaySupport  [ sway makeWrapper ];
```

Comment:
You can take `lib` as an input and save yourself writing `stdenv.` over and over.